app/services/requirements/image_checker.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Requirements
+  class ImageChecker
+    ALT_TEXT_MAX_LENGTH = 125
+    CAPTION_MAX_LENGTH = 160
+
+    attr_reader :image
+
+    def initialize(image)
+      @image = image
+    end
+
+    def pre_draft_metadata_issues
+      issues = []
+
+      if image.alt_text.blank?
+        issues << Issue.new(:alt_text, :blank, filename: image.filename)
+      end
+
+      if image.alt_text.to_s.length > ALT_TEXT_MAX_LENGTH
+        issues << Issue.new(:alt_text, :too_long, max_length: ALT_TEXT_MAX_LENGTH, filename: image.filename)
+      end
+
+      if @image.caption.to_s.length > CAPTION_MAX_LENGTH
+        issues << Issue.new(:caption, :too_long, max_length: CAPTION_MAX_LENGTH, filename: image.filename)
+      end
+
+      CheckerIssues.new(issues)
+    end
+
+    def pre_draft_issues
+      pre_draft_metadata_issues
+    end
```

Comment:
I think this would be better if we just had a single `pre_draft_issues` method here as I found the metadata aspect to be a bit confusing. My thought in https://github.com/alphagov/content-publisher/pull/485/files#diff-3fc221a1b655a04cb0f87aa766b94161R74 was "why do we only care about metadata issues" ? 

I'm assuming you've done this with consistency with DocumentChecker but I think these classes are somewhat different as that class has a wider responsibility of the document - whereas this is focussed on images

Reply:
Resolved as discussed - the update action should check specifically for metadata issues, as this is what it's updating.